pol-core/pol/uworld.h

```diff
@@ -104,7 +97,18 @@ inline Zone& getzone( unsigned short x, unsigned short y, Realms::Realm* realm )
   passert( x < realm->width() );
   passert( y < realm->height() );
 
-  return realm->zone[x >> Plib::WGRID_SHIFT][y >> Plib::WGRID_SHIFT];
+  return realm->getzone( x >> Plib::WGRID_SHIFT, y >> Plib::WGRID_SHIFT );
+}
+
+inline Pos2d zone_convert( const Pos4d& p )
+{
+  return Pos2d( static_cast<unsigned short>( p.x() >> Plib::WGRID_SHIFT ),
+                static_cast<unsigned short>( p.y() >> Plib::WGRID_SHIFT ) );
+}
+
+inline Zone& getzone( const Pos4d& p )
```

Comment:
Just to remember that getzone(x,y) and realm->getzone(x,y) are using different units. The first is world location, the second is grid coordinate. Maybe we can think of a better name to distinguish them